tests/unit/processing/test_spatial_assets_utils.py

```diff
@@ -0,0 +1,315 @@
+import copy
+import os
+import pickle
+import tempfile
+from os import mkdir
+from uuid import uuid4
+
+import numpy as np
+import pytest
+from PIL import Image
+
+from backend.layers.processing.utils.cxg_generation_utils import convert_uns_to_cxg_group
+from backend.layers.processing.utils.spatial import SpatialDataProcessor
+from backend.layers.thirdparty.s3_provider import S3Provider
+from tests.unit.backend.fixtures.environment_setup import fixture_file_path
+
+
+@pytest.fixture
+def setup_spatial_processor(mocker):
+    testing_cxg_temp_directory = fixture_file_path(str(uuid4()))
+
+    s3_provider_mock = mocker.MagicMock(spec=S3Provider)
+    library_id = "abcd123"
+    valid_uns = {
+        "spatial": {
+            "library_id_1": {
+                "images": {"hires": np.random.rand(10, 10, 3), "fullres": np.random.rand(20, 20, 3)},
+                "scalefactors": {"spot_diameter_fullres": 1.0, "tissue_hires_scalef": 0.5},
+            }
+        }
+    }
+    valid_spatial_data = valid_uns["spatial"]["library_id_1"]
+    invalid_uns = {}
+    spatial_processor = SpatialDataProcessor(s3_provider=s3_provider_mock)
+    mkdir(testing_cxg_temp_directory)
+
+    output_folder = "test_output"
+    cxg_container = "test_container"
+    group_metadata_name = "uns"
+    ctx = None
+
+    mock_spatial_processor = mocker.MagicMock(spec=SpatialDataProcessor)
+    mock_spatial_processor.filter_spatial_data.return_value = valid_uns["spatial"]
+
+    return {
```

Comment:
Is all of this setup needed for each tests? For reusability could you break this down into small fixture: 1 to reduce the setup overhead for each test, and to make it clear exactly what fixture are need for each test to run successfully.